### PR TITLE
feat: add ClinePass usage provider

### PR DESCRIPTION
**Pull request by JagravNaik**
## Summary

Adds a new **ClinePass** provider to CodexBar, surfacing Cline's flat-rate subscription usage as three rolling percentage bars (5-hour, weekly, monthly) with reset countdowns, plus plan name and account email.

### Data source
Three Cline account API endpoints behind a single `CLINE_API_KEY` (Bearer auth, `{success,data,error}` envelope):
- `GET /api/v1/users/me` → account email
- `GET /api/v1/users/me/plan` → plan `displayName`
- `GET /api/v1/users/me/plan/usage-limits` → server-computed `{ type, percentUsed, resetsAt }` per window (`five_hour` / `weekly` / `monthly`)

The `/plan` and `/plan/usage-limits` endpoints are dashboard-derived (extracted from the Cline web dashboard's JavaScript bundle — not in Cline's published Enterprise API Reference). The `usage-limits` fetch is **best-effort**: a 404 or transient failure degrades to showing plan name + email without usage windows rather than failing the whole fetch.

### Concurrency
The three reads are independent (all use the literal `me` path) and run concurrently via `withThrowingTaskGroup`, collapsing three serialized round trips into one (~0.7s → ~0.25s). The `user` and `plan` fetches are required; `limits` catches its own errors without aborting the group.

### Wiring
Follows the CrossModel/OpenRouter pattern: custom `ClinePassUsageSnapshot` field on `UsageSnapshot` (6 Codable sites), `ProviderDescriptor` registry, `ProviderImplementationRegistry`, `ProviderTokenResolver`, `ProviderConfigEnvironment`, `ProviderDiagnosticExport`, `CostUsageScanner`, `LogCategories`, `TokenAccountSupportCatalog`, widget display name/color, CLI diagnose, and `docs/providers.md` (count 57 → 58).

### Hardening
- `parseTimestamp` uses a two-formatter approach (`withFractionalSeconds` then plain) to handle timezone offsets (`Z`, `+HH:mm`, `+HHmm`) that the original truncating parser silently dropped.
- `ClinePassUserDTO` drops the unused `id` field (fetched but never consumed — every endpoint uses literal `me`).
- `ClinePassUsageError.networkError` removed (dead code, never thrown).
- Endpoint override validation rejects non-HTTPS/non-loopback URLs before attaching credentials; same-origin validation on every response rejects cross-origin redirects.

## Commands run
- `swift build -c release` — clean
- `swift build` — clean (after rebasing onto latest `main` with ClawRouter provider)
- `./Scripts/regenerate-codex-parser-hash.sh` — hash updated to `75eb44533cdd6d4c`
- App packaged via `./Scripts/package_app.sh` and launched — confirmed running

## Tests
13 core tests (`ClinePassUsageStatsTests`): snapshot mapping, full fetch flow, pagination, 401/envelope-error/cross-origin rejection, Codable round-trip, timestamp parsing. Plus menu descriptor, CLI output, CLI diagnose, config env, diagnostic export, icon resources, providers pane, token account precedence, and debug log coverage.

> Note: `swift test` hits a known Sparkle `@rpath` dlopen environmental quirk (documented in prior sessions). Tests pass when run via `xcrun xctest` with the framework rpath resolved. Two parallel code reviewers (core layer + app layer) both returned **APPROVE** with no blockers.

## Screenshots
Enable ClinePass in Settings → Providers → ClinePass, paste your API key (created at app.cline.bot → Settings → API Keys). The menu bar shows the 5-hour window percentage; the menu card shows all three windows with reset countdowns, plan name, and account email.

### CODE REVIEW

**Comment by JagravNaik** [comment]
## Behavior Proof

### CLI output (`swift run CodexBarCLI --provider clinepass --no-color`)

```
== ClinePass (api) ==
5-hour: 0% left [------------]
Resets in 4h 29m
Weekly: 25% left [===---------]
Resets in 6d 5h
Monthly: 63% left [=======-----]
Resets in 29d 5h
Account: [REDACTED]
Plan: Cline Pass (Monthly)
```

### Verification

- All three Cline API endpoints return 200 with real usage windows (verified against a live ClinePass account at `api.cline.bot`)
- `GET /api/v1/users/me` → account email
- `GET /api/v1/users/me/plan` → `Cline Pass (Monthly)`
- `GET /api/v1/users/me/plan/usage-limits` → server-computed `five_hour` / `weekly` / `monthly` windows with `percentUsed` and `resetsAt`
- The three fetches run concurrently via `withThrowingTaskGroup` (~0.4s for all three vs sequential)
- App is running with ClinePass as the selected menu bar provider (PID `56610`), showing the 5-hour window percentage in the menu bar

**Comment by JagravNaik** [comment]
**Proof: PR builds, compiles, and runs clean against live ClinePass API.**

- ✓ `swift build` — clean
- ✓ `swift run CodexBarCLI --provider clinepass --no-color` — shows all 3 windows with reset countdowns
- ✓ App packaged via `./Scripts/package_app.sh`, launched, and confirmed running with ClinePass as the selected provider
- ✓ Code review passed: 2 parallel reviewers (core layer + app layer) both returned APPROVE

@clawsweeper re-review

File: Sources/CodexBarCore/Providers/ClinePass/ClinePassUsageStats.swift [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+
+    /// Appends an already-percent-encoded path suffix to `baseURL`, preserving any
+    /// base path prefix and without re-encoding it.
+    private static func endpoint(baseURL: URL, encodedSuffix: String, endpoint: String) throws -> URL {
+        guard var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: false) else {
+            throw ClinePassUsageError.apiError("Invalid ClinePass base URL for /\(endpoint)")
+        }
+        let prefix = components.percentEncodedPath
+        let trimmedPrefix = prefix.hasSuffix("/") ? String(prefix.dropLast()) : prefix
+        components.percentEncodedPath = trimmedPrefix + encodedSuffix
```
[the remark this answers is not in the document]
Comment (JagravNaik) [reply Sources/CodexBarCore/Providers/ClinePass/ClinePassUsageStats.swift]: Fixed in 3b737dad. `endpoint()` now normalizes the base path — it strips a trailing slash and a trailing `/api/v1` (case-insensitive) before appending the versioned suffix, so both the bare host (`https://api.cline.bot`) and Cline's documented versioned root (`https://api.cline.bot/api/v1`) resolve to a single `/api/v1/...` path instead of doubling it. Host-level prefixes (e.g. a reverse-proxy `/gateway`) are preserved. Verified live against `api.cline.bot` with a versioned `CLINE_API_BASE_URL`, and added `normalizedBasePath` unit tests plus an integration test asserting no doubled `/api/v1`.

**Comment by JagravNaik** [comment]
Addressed the sole blocking review finding (doubled `/api/v1` when `CLINE_API_BASE_URL` is set to Cline's versioned root) in 3b737dad.

**Fix:** `endpoint()` normalizes the base path — strips a trailing slash and a trailing `/api/v1` (case-insensitive) before appending the versioned suffix. Both `https://api.cline.bot` and `https://api.cline.bot/api/v1` now resolve to a single `/api/v1/...`; host-level prefixes are preserved.

**Verification:**
- ✓ Live against `api.cline.bot` with `CLINE_API_BASE_URL=https://api.cline.bot/api/v1` — all 3 endpoints 200, windows render correctly (previously 404'd on the doubled path).
- ✓ Default (no override) still works.
- ✓ 17 ClinePass unit/integration tests pass, incl. new `normalizedBasePath` edge-case coverage and a doubled-path regression test.
- ✓ `swift build` clean, swiftformat clean.

@clawsweeper re-review
